src/plugins/cache.js

```diff
@@ -107,6 +127,21 @@ export default class Cache {
 
   async fetchManagers (...managers: Manager[]): Promise<Plugin[]> {
     let plugins = []
+    if (this.cache.node_version !== process.version) {
+      let lock = new Lock(this.out)
+      let handled = true
+
+      let downgrade = await lock.upgrade()
+      for (let manager of managers) {
+        handled = await manager.handleNodeVersionChange() && handled
```

Comment:
for now, I'd like to just fail with the warning without attempting to rebuild. If we start to see issues where users are having rebuilds that fail once then work again we should revisit. In that case it might also start to make sense to store the node version alongside each plugin as well since we wouldn't have to keep the top-level index aware of individual failures or have to retry all the plugins if it fails